libs/wingc/src/type_check/jsii_importer.rs

```diff
@@ -172,47 +147,54 @@ impl<'a> JsiiImporter<'a> {
 			.unwrap()
 	}
 
-	fn import_type(&mut self, type_fqn: &FQN) {
+	pub fn import_type(&mut self, type_fqn: &FQN) {
 		// Hack: if the class name is a construct base then we treat this class as a resource and don't need to define it
 		if is_construct_base(&type_fqn) {
 			return;
 		}
 
+		let type_str = type_fqn.as_str();
+
 		self.setup_namespaces_for(&type_fqn);
 
 		// Hack: if the type is "constructs.IConstruct", we import it manually
 		// this is done so we can avoid loading the constructs module
-		if type_fqn.as_str() == CONSTRUCT_BASE_INTERFACE {
+		if type_str == CONSTRUCT_BASE_INTERFACE {
 			let symbol = Symbol::global(type_fqn.type_name());
 			self.register_jsii_type(&type_fqn, &symbol, self.wing_types.anything());
 			return;
 		}
 
+		// check if type is already imported
+		if self.wing_types.libraries.lookup_nested_str(type_str, None).is_ok() {
+			return;
+		}
+
 		// Check if this is a JSII interface and import it if it is
-		let jsii_interface = self.jsii_types.find_interface(type_fqn);
+		let jsii_interface = self.jsii_spec.type_system.find_interface(type_fqn);
 		if let Some(jsii_interface) = jsii_interface {
 			self.import_interface(jsii_interface);
 			return;
 		}
 
 		// Check if this is a JSII class and import it if it is
-		let jsii_class = self.jsii_types.find_class(type_fqn);
+		let jsii_class = self.jsii_spec.type_system.find_class(type_fqn);
 		if let Some(jsii_class) = jsii_class {
 			self.import_class(jsii_class);
 			return;
 		}
 
 		// Check if this is a JSII enum and import it if it is
-		let jsii_enum = self.jsii_types.find_enum(type_fqn);
+		let jsii_enum = self.jsii_spec.type_system.find_enum(type_fqn);
 		if let Some(jsii_enum) = jsii_enum {
 			self.import_enum(jsii_enum);
 			return;
 		}
 
-		panic!("Type {} was not found in the type system", type_fqn);
+		debug!("Type {} was not found in the type system", type_fqn);
```

Comment:
Not sure about this one. This function is really just responsible for updating the internal collection of types. The real error that surfaces to the user later is that the type isn't found. I don't feel like adding an additional diagnostic here gives users any more info.

Reply:
> in theory aren't we always importing known types?

Not anymore, if the user tries `cloud.butt` it will try to find it in the assembly via this method.

I can convert this method to have a boolean return type (Result would be overkill IMO) to make the intention more clear.